website/project/decorators.py

```diff
@@ -41,19 +41,43 @@ def _kwargs_to_nodes(kwargs):
     return project, node
 
 
-def must_be_valid_project(func):
+def must_be_valid_project(func=None, are_retractions_valid=False):
+    """ Ensures permissions to retractions are never implicitly granted. """
 
     # TODO: Check private link
+    def must_be_valid_project_inner(func):
+
+        @functools.wraps(func)
+        def wrapped(*args, **kwargs):
+
+            kwargs['project'], kwargs['node'] = _kwargs_to_nodes(kwargs)
+            if not are_retractions_valid and kwargs['project'].is_retracted:
```

Comment:
I wasn't sure the best way to write the conditional (components are often None and I didn't want to write a hideous conditional) but I did it anyway. Let me know what you think in the follow up